### DISCUSSION TITLE
Animating the theme change?

### DISCUSSION THREAD

**Discussion by AgarwalPragy** [saadeghi/daisyui#1207] [Q&A] [answered]
I'm trying to achieve an effect like this
- https://www.youtube.com/watch?v=MhEPd1HoVH8 
- https://www.youtube.com/watch?v=oIHuvg38qLE

As per the docs, the theme can be set via the `data-theme` attribute

```html
<html data-theme="cupcake"></html>
```

When the attribute changes, the theme on the page changes immediately. Is there a way of achieving the same effect with daisyui?

**Comment by saadeghi** [comment, marked as answer]
This is challenging. During the animation, we need to see the same page content with both themes. Some part of the content is inside the circle and the rest is outside. And it's not about inverting all colors
 Each color may have its own value on each theme so the cricle can't be only a color filter to invert all colors.

That being said, I think the practical way to do this would be duplicating the whole page (everything inside the container (probably `<body>`). Having the second theme for the clone. and then use the circle as a mask to show the second theme as the cricle grows.

It's a cool effect but I personally would never do this on production. It's just not efficient and cloning the whole page may cause other problems.

If anyone could come up with a better solution, please share it with us.

**Reply by AgarwalPragy** [reply]
> It's a cool effect but I personally would never do this on production. It's just not efficient and cloning the whole page may cause other problems.

Agreed. Maybe I can take a screenshot of the page, put it in the foreground, change the theme and slowly reveal the page from under the image. Will post back if that works.

Thanks :)

**Reply by saadeghi** [reply]
Well, I don't think talking screenshot is any better because the page is (probably) responsive and the content is (probably) dynamic.

**Reply by AgarwalPragy** [reply]
I meant taking a screenshot when the theme toggle is activated. While the theme animation is going on, the page being inactive for 2s is okay I guess.

**Reply by AgarwalPragy** [reply]
@saadeghi PS: I've sent you an email regarding freelancing. Please take a look when you have time ^_^
